### PR TITLE
Resolves #226 "Transition from javax to jakarta"

### LINKED ISSUE #226: Transition from javax to jakarta
The transition from the "javax" namespace to "Jakarta" in the Java ecosystem is a significant development with far-reaching implications.

This issue aims to move JWPL to "jakarta.*".

JWPL mainly relies on two EE specs, which require some work in order to sort it out:

- JPA through Hibernate
- JAXB through transient dependencies (mainly sweble engine)

As this will break consumers (at least at runtime), it needs to come with a major version bump

### PR DESCRIPTION
**Pull request by rzo1**
# What does this PR do?

This pull request brings compatibility to the Jakarta EE 9.1 Spec. Due to the namespace change this might break consumers, so I increased the version to 2.0.0-SNAPSHOT. 

Since Sweble is not maintained anymore and has a hard binding against an old JAXB, I decided to use a shading approach for the javax -> jakarta migration (like we do in numerous ASF projects with EE reference). 

During the upgrade I did several dependency upgrades or cleanups. For example, I removed the hard dependencies towards ant or trove4j. Either by forking (ant bzip2 code) or by replacing with stock JDK code. This might impact performance, but might still be better than relying on unmaintained libs. 

We ran the performance tests locally and the results are in #226, so in principle the code stack still works after the changes.

Happy to receive some feedback.

### CODE REVIEW

File: de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java
Comment (rzo1) [inline de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: Might be worth to shade that in `deps` as well to remove code duplication and ease possible updates? wdyt?

**Review by mawiesne — APPROVED** [review]
Thanks for all the effort and the progress that this PR will bring.

**Comment by rzo1** [comment]
@reckart Can we reconfigure the Jenkins for this repo? The PR introduces GH actions, but it seems, that this repo still relies on a decomissioned (?) build instance ?

**Comment by reckart** [comment]
I have removed the jenkins check from the master branch - is that what you were asking for?

**Comment by rzo1** [comment]
> I have removed the jenkins check from the master branch - is that what you were asking for?

Yes ;-)

**Comment by reckart** [comment]
Hm, considering that the PR includes a github action - shouldn't this run automatically? Is there anything we need to do for it to activate?

**Comment by rzo1** [comment]
> Hm, considering that the PR includes a github action - shouldn't this run automatically? Is there anything we need to do for it to activate?

I think, that the GH action needs to be added on "main" first, I guess (or activated for the repo/org maybe?).

**Comment by rzo1** [comment]
Yeah, I have a similar impression. They are running on the fork: https://github.com/rzo1/dkpro-jwpl/actions/runs/6559026682/job/17813774348 - so an organization setting maybe?

**Comment by reckart** [comment]
I'll tentatively approve to PR - let's see if that helps something...

**Review by reckart — APPROVED** [review]

**Comment by reckart** [comment]
The repo settings say that first-time contributors need approval - but the PR already has two approvals now...

**Comment by rzo1** [comment]
Ok, now it is running ;-)

**Comment by reckart** [comment]
Merged the other branch to main - let's see if this then also helps here... there are some conflicts now.

**Comment by rzo1** [comment]
Just rebased this PR, so should be ok now in terms of conflicts

File: de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/WikiHibernateUtil.java
Comment (reckart) [inline de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/WikiHibernateUtil.java:115]: Having useMariaDB in both conditions seems pointless.

File: de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java
Comment (reckart) [reply de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: Do you have an idea why the code is copied anyway? Why not have an actual dependency?

File: de.tudarmstadt.ukp.wikipedia.parser/pom.xml
Comment (reckart) [inline de.tudarmstadt.ukp.wikipedia.parser/pom.xml:47]: Why provided instead of test?

File: de.tudarmstadt.ukp.wikipedia.revisionmachine/src/main/java/de/tudarmstadt/ukp/wikipedia/revisionmachine/difftool/config/gui/panels/FilterPanel.java
Comment (reckart) [inline de.tudarmstadt.ukp.wikipedia.revisionmachine/src/main/java/de/tudarmstadt/ukp/wikipedia/revisionmachine/difftool/config/gui/panels/FilterPanel.java:202]: Why not simply `true` and `false`?

File: de.tudarmstadt.ukp.wikipedia.timemachine/src/main/java/de/tudarmstadt/ukp/wikipedia/timemachine/dump/version/DumpVersionTroveIntKey.java
Comment (reckart) [inline de.tudarmstadt.ukp.wikipedia.timemachine/src/main/java/de/tudarmstadt/ukp/wikipedia/timemachine/dump/version/DumpVersionTroveIntKey.java:48]: Wouldn't using something like FastUtil be reasonable here?

File: deps/pom.xml
Comment (reckart) [inline deps/pom.xml:26]: IMHO artifact names should all share a common prefix like e.g. `jwpl`.

File: pom.xml
Comment (reckart) [inline pom.xml:109]: Isn't runtime transitive? Do we want that? Normally, dependencies in dependency management should not have a scope.

File: de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java
Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: I don't think, that a library should rely on a build tool (here: ant) with all of it's stuff if we only need 5 classes related to bzip2 processing.

Comment (reckart) [reply de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: Nothing related in commons-compress?

Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: I copied / forked it to discuss this issue within this PR. I would be in favour of only shading that 5 classes in `/deps` and re-use it over the code base.

File: deps/pom.xml
Comment (rzo1) [reply deps/pom.xml:26]: I have no objections in changing to `dkpro-jwpl-XXX`. WDYT @mawiesne ?

File: de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java
Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: Let me check commons-compress

File: deps/pom.xml
Comment (reckart) [reply deps/pom.xml:26]: And while at it, the groupId could also be updated to `org.dkpro.jwpl`.

Comment (reckart) [reply deps/pom.xml:26]: (e.g. DKPro Core as groupId `org.dkpro.core` for some time now)

File: de.tudarmstadt.ukp.wikipedia.parser/pom.xml
Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.parser/pom.xml:47]: Think it is just CnP from my side.

It seems, that `parser` is intended as a library module. Other modules such as "dumper", etc. are more or less CLI modules (with provided log4j2 deps). Will fix it.

File: deps/pom.xml
Comment (mawiesne) [reply deps/pom.xml]: see: https://github.com/dkpro/dkpro-jwpl/issues/165

Comment (mawiesne) [reply deps/pom.xml]: I agree. we could do the relocation / renaming as proposed in #165, but separately.

Comment (rzo1) [reply deps/pom.xml]: Yeah - let's tackle this for 2.0.0 but in another PR to ease reviewer burden :)

**Comment by mawiesne** [comment]
@reckart Could you please de-activate the Jenkins hook from this project's settings entirely? The other open PR is also blocked merely by Jenkins integration still interfering...

File: de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java
Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:1]: looks good. removed forked code in favour of commons-compress.

**Comment by mawiesne** [comment]
> I have removed the jenkins check from the master branch - is that what you were asking for?

Please remove it project-wide, that is, from all branches / PRs. Thx in advance.

File: de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/WikiHibernateUtil.java
Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/WikiHibernateUtil.java:115]: Looked in the history. Looks like a refactoring error and should be hsqldb. I will fix it. Nice catch!

File: de.tudarmstadt.ukp.wikipedia.timemachine/src/main/java/de/tudarmstadt/ukp/wikipedia/timemachine/dump/version/DumpVersionTroveIntKey.java
Comment (rzo1) [reply de.tudarmstadt.ukp.wikipedia.timemachine/src/main/java/de/tudarmstadt/ukp/wikipedia/timemachine/dump/version/DumpVersionTroveIntKey.java]: Switched to fastutil-core

**Comment by rzo1** [comment]
@reckart Addressed most of your comments. I would like to do the renaming of artifacts + group-ids in https://github.com/dkpro/dkpro-jwpl/issues/165 after this PR is merged.

**Review by mawiesne — APPROVED** [review]

**Comment by reckart** [comment]
> Please remove it project-wide, that is, from all branches / PRs. Thx in advance.

It is removed, but if there are old PRs that still have a failed state from Jenkins, I'm afraid GitHub doesn't seem to be smart enough to remove those... ping me in such cases, I can force-merge them.

**Comment by reckart** [comment]
@rzo1 So, are we good to merge?

**Comment by reckart** [comment]
@rzo1 then feel free to hit the button

**Pull request merged**
